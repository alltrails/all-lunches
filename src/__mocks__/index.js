export { default as restaurantMock } from './restaurant.mock';
export { default as restaurantsMock } from './restaurants.mock';
export { default as reactRefMock } from './reactRef.mock';
