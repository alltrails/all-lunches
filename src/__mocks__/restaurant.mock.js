/* eslint-disable sort-keys  */

export default {
  coordinates: [-122.408999, 37.7902566],
  id: 'ChIJkVLoONZ_j4ARx2vJ4umERnI',
  name: "Uncle Vito's Pizza",
  photoUrl:
    'https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623',
  placeUrl:
    "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
  priceLevel: 1,
  rating: 4.3,
  supportingText: 'restaurant',
  userRatingsTotal: 1427,
};
