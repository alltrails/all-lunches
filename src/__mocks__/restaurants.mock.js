import DefaultStorePNG from 'assets/store-front.png';

export default [
  {
    coordinates: [-122.408999, 37.7902566],
    id: 'ChIJkVLoONZ_j4ARx2vJ4umERnI',
    name: "Uncle Vito's Pizza",
    photoUrl: DefaultStorePNG,
    placeUrl:
      "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
    priceLevel: 1,
    rating: 4.3,
    supportingText: 'restaurant',
    userRatingsTotal: 1427,
  },
  {
    coordinates: [-122.408675, 37.7900735],
    id: 'ChIJg7cNIYyAhYAR8D4ml1oItTg',
    name: "Roxanne's Cafe",
    photoUrl: DefaultStorePNG,
    placeUrl:
      "https://www.google.com/maps/search/?api=1&query=Roxanne's%20Cafe&query_place_id=ChIJg7cNIYyAhYAR8D4ml1oItTg",
    priceLevel: 2,
    rating: 4.2,
    supportingText: 'restaurant',
    userRatingsTotal: 1219,
  },
  {
    coordinates: [-122.4073542, 37.7902509],
    id: 'ChIJaa9OAYyAhYARxwgUHLNEpOY',
    name: 'Bouche',
    photoUrl: DefaultStorePNG,
    placeUrl:
      'https://www.google.com/maps/search/?api=1&query=Bouche&query_place_id=ChIJaa9OAYyAhYARxwgUHLNEpOY',
    priceLevel: 3,
    rating: 4.5,
    supportingText: 'bar',
    userRatingsTotal: 465,
  },
  {
    coordinates: [-122.409088, 37.7902449],
    id: 'ChIJvXXIOoyAhYARe-ixq3WdmrY',
    name: 'Sons & Daughters',
    photoUrl: DefaultStorePNG,
    placeUrl:
      'https://www.google.com/maps/search/?api=1&query=Sons%20%26%20Daughters&query_place_id=ChIJvXXIOoyAhYARe-ixq3WdmrY',
    priceLevel: 4,
    rating: 4.6,
    supportingText: 'restaurant',
    userRatingsTotal: 249,
  },
];
