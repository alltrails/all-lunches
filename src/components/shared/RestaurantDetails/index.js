/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';

import DefaultStorePNG from 'assets/store-front.png';
import StarIcon from 'styles/components/StarIcon';
import {
  MetadataWrapper,
  PlaceImage,
  PriceLevelWrapper,
  RatingStarsWrapper,
  Title,
  HyperLinkWrapper,
  UserRatingsTotalText,
  StarIconWrapper,
} from './style';

const RestaurantDetails = ({
  name,
  photoUrl,
  placeUrl,
  priceLevel,
  rating,
  supportingText,
  userRatingsTotal,
}) => {
  const getPriceLevel = (price) => {
    if (!price) return 'N/A';
    let dollarSigns = '';

    for (let i = 0; i < priceLevel; i += 1) dollarSigns += '$';
    return dollarSigns;
  };

  const placeImage = photoUrl || DefaultStorePNG;
  return (
    <HyperLinkWrapper href={placeUrl} target="_blank" rel="noopener noreferrer">
      <PlaceImage src={placeImage} alt="storefront" />
      <MetadataWrapper>
        <Title>{name}</Title>
        <RatingStarsWrapper>
          {[...Array(5)].map((_, index) => (
            <StarIconWrapper key={index}>
              <StarIcon color={index + 1 <= Math.round(rating) ? '#f5d24a' : '#E6E6E6'} />
            </StarIconWrapper>
          ))}
          <UserRatingsTotalText>({userRatingsTotal})</UserRatingsTotalText>
        </RatingStarsWrapper>
        <PriceLevelWrapper>
          <span>{getPriceLevel(priceLevel)} </span>
          <span> {'\u2022'} </span>
          <span> {supportingText}</span>
        </PriceLevelWrapper>
      </MetadataWrapper>
    </HyperLinkWrapper>
  );
};

RestaurantDetails.propTypes = {
  name: PropTypes.string,
  photoUrl: PropTypes.string,
  placeUrl: PropTypes.string,
  priceLevel: PropTypes.number,
  rating: PropTypes.number,
  supportingText: PropTypes.string,
  userRatingsTotal: PropTypes.number,
};

RestaurantDetails.defaultProps = {
  name: 'N/A',
  photoUrl: null,
  priceLevel: 0,
  rating: 'N/A',
  userRatingsTotal: 0,
  placeUrl: null,
  supportingText: 'N/A',
};

export default RestaurantDetails;
