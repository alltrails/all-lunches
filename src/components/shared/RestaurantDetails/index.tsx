/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';

import DefaultStorePNG from 'assets/store-front.png';
import StarIcon from 'styles/components/StarIcon';
import {
  MetadataWrapper,
  PlaceImage,
  PriceLevelWrapper,
  RatingStarsWrapper,
  Title,
  HyperLinkWrapper,
  UserRatingsTotalText,
  StarIconWrapper,
} from './style';

type RestaurantDetailsProps = {
  name?: string;
  photoUrl?: string;
  placeUrl?: string;
  priceLevel?: number;
  rating?: number;
  supportingText?: string;
  userRatingsTotal?: number;
};

const RestaurantDetails = ({
  name = 'N/A',
  photoUrl = '',
  placeUrl = '',
  priceLevel = 0,
  rating = 0,
  supportingText = 'N/A',
  userRatingsTotal = 0,
}: RestaurantDetailsProps) => {
  const getPriceLevel = (price: number) => {
    if (!price) return 'N/A';
    let dollarSigns = '';

    for (let i = 0; i < priceLevel; i += 1) dollarSigns += '$';
    return dollarSigns;
  };

  const placeImage = photoUrl || DefaultStorePNG;
  return (
    <HyperLinkWrapper href={placeUrl} target="_blank" rel="noopener noreferrer">
      <PlaceImage src={placeImage} alt="storefront" />
      <MetadataWrapper>
        <Title>{name}</Title>
        <RatingStarsWrapper>
          {[...Array(5)].map((_, index) => (
            <StarIconWrapper key={index}>
              <StarIcon color={index + 1 <= Math.round(rating) ? '#f5d24a' : '#E6E6E6'} />
            </StarIconWrapper>
          ))}
          <UserRatingsTotalText>({userRatingsTotal})</UserRatingsTotalText>
        </RatingStarsWrapper>
        <PriceLevelWrapper>
          <span>{getPriceLevel(priceLevel)} </span>
          <span> {'\u2022'} </span>
          <span> {supportingText}</span>
        </PriceLevelWrapper>
      </MetadataWrapper>
    </HyperLinkWrapper>
  );
};

RestaurantDetails.propTypes = {
  name: PropTypes.string,
  photoUrl: PropTypes.string,
  placeUrl: PropTypes.string,
  priceLevel: PropTypes.number,
  rating: PropTypes.number,
  supportingText: PropTypes.string,
  userRatingsTotal: PropTypes.number,
};

RestaurantDetails.defaultProps = {
  name: 'N/A',
  photoUrl: null,
  priceLevel: 0,
  rating: 0,
  userRatingsTotal: 0,
  placeUrl: null,
  supportingText: 'N/A',
};

export default RestaurantDetails;
