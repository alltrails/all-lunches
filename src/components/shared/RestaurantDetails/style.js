import styled from 'styled-components';
import { colors, type } from 'styles';

export const HyperLinkWrapper = styled.a`
  text-decoration: none;
  outline: none;
  display: flex;
  flex-direction: row;
  text-decoration: none;
`;

export const PlaceImage = styled.img`
  width: 65px;
  height: 65px;
  margin-right: 10px;
`;

export const MetadataWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
`;

export const Title = styled.div`
  color: ${colors.grayDark};
  font-size: 1.4rem;
  font-weight: ${type.weight.bold};
  padding-right: 6px;
`;

export const RatingStarsWrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const StarIconWrapper = styled.i`
  margin-right: 2px;
`;

export const PriceLevelWrapper = styled.div`
  font-size: 1.2rem;
  color: ${colors.textLightGray};
  text-transform: capitalize;
`;

export const UserRatingsTotalText = styled.span`
  font-size: 1.2rem;
  color: ${colors.spanLightGray};
  margin-left: 5px;
`;
