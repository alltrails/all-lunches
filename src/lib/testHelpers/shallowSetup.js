import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
