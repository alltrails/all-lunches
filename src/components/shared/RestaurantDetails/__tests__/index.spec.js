/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';
import { restaurantMock } from '__mocks__';

import RestaurantDetails from '../index';

describe('RestaurantDetails', () => {
  const defaultProps = {};
  const setup = shallowSetup(RestaurantDetails, defaultProps);

  it('renders the expected view when no properties exist', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when all properties exist', () => {
    const { wrapper } = setup({
      name: restaurantMock.name,
      photoUrl: restaurantMock.photoUrl,
      placeUrl: restaurantMock.placeUrl,
      priceLevel: restaurantMock.priceLevel,
      rating: restaurantMock.rating,
      supportingText: restaurantMock.supportingText,
      userRatingsTotal: restaurantMock.userRatingsTotal,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when price level is high', () => {
    const { wrapper } = setup({
      name: restaurantMock.name,
      photoUrl: restaurantMock.photoUrl,
      placeUrl: restaurantMock.placeUrl,
      priceLevel: 5,
      rating: restaurantMock.rating,
      supportingText: restaurantMock.supportingText,
      userRatingsTotal: restaurantMock.userRatingsTotal,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when rating level is low', () => {
    const { wrapper } = setup({
      name: restaurantMock.name,
      photoUrl: restaurantMock.photoUrl,
      placeUrl: restaurantMock.placeUrl,
      priceLevel: restaurantMock.priceLevel,
      rating: 0,
      supportingText: restaurantMock.supportingText,
      userRatingsTotal: restaurantMock.userRatingsTotal,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
