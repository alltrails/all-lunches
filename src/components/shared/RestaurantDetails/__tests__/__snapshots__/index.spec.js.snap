// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantDetails renders the expected view when all properties exist 1`] = `
<style__HyperLinkWrapper
  href="https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI"
  rel="noopener noreferrer"
  target="_blank"
>
  <style__PlaceImage
    alt="storefront"
    src="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623"
  />
  <style__MetadataWrapper>
    <style__Title>
      Uncle Vito's Pizza
    </style__Title>
    <style__RatingStarsWrapper>
      <style__StarIconWrapper
        key="0"
      >
        <StarIcon
          color="#f5d24a"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="1"
      >
        <StarIcon
          color="#f5d24a"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="2"
      >
        <StarIcon
          color="#f5d24a"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="3"
      >
        <StarIcon
          color="#f5d24a"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="4"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__UserRatingsTotalText>
        (
        1427
        )
      </style__UserRatingsTotalText>
    </style__RatingStarsWrapper>
    <style__PriceLevelWrapper>
      <span>
        $
         
      </span>
      <span>
         
        •
         
      </span>
      <span>
         
        restaurant
      </span>
    </style__PriceLevelWrapper>
  </style__MetadataWrapper>
</style__HyperLinkWrapper>
`;

exports[`RestaurantDetails renders the expected view when no properties exist 1`] = `
<style__HyperLinkWrapper
  href={null}
  rel="noopener noreferrer"
  target="_blank"
>
  <style__PlaceImage
    alt="storefront"
    src="test-file-stub"
  />
  <style__MetadataWrapper>
    <style__Title>
      N/A
    </style__Title>
    <style__RatingStarsWrapper>
      <style__StarIconWrapper
        key="0"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="1"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="2"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="3"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="4"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__UserRatingsTotalText>
        (
        0
        )
      </style__UserRatingsTotalText>
    </style__RatingStarsWrapper>
    <style__PriceLevelWrapper>
      <span>
        N/A
         
      </span>
      <span>
         
        •
         
      </span>
      <span>
         
        N/A
      </span>
    </style__PriceLevelWrapper>
  </style__MetadataWrapper>
</style__HyperLinkWrapper>
`;

exports[`RestaurantDetails renders the expected view when price level is high 1`] = `
<style__HyperLinkWrapper
  href="https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI"
  rel="noopener noreferrer"
  target="_blank"
>
  <style__PlaceImage
    alt="storefront"
    src="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623"
  />
  <style__MetadataWrapper>
    <style__Title>
      Uncle Vito's Pizza
    </style__Title>
    <style__RatingStarsWrapper>
      <style__StarIconWrapper
        key="0"
      >
        <StarIcon
          color="#f5d24a"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="1"
      >
        <StarIcon
          color="#f5d24a"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="2"
      >
        <StarIcon
          color="#f5d24a"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="3"
      >
        <StarIcon
          color="#f5d24a"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="4"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__UserRatingsTotalText>
        (
        1427
        )
      </style__UserRatingsTotalText>
    </style__RatingStarsWrapper>
    <style__PriceLevelWrapper>
      <span>
        $$$$$
         
      </span>
      <span>
         
        •
         
      </span>
      <span>
         
        restaurant
      </span>
    </style__PriceLevelWrapper>
  </style__MetadataWrapper>
</style__HyperLinkWrapper>
`;

exports[`RestaurantDetails renders the expected view when rating level is low 1`] = `
<style__HyperLinkWrapper
  href="https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI"
  rel="noopener noreferrer"
  target="_blank"
>
  <style__PlaceImage
    alt="storefront"
    src="https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623"
  />
  <style__MetadataWrapper>
    <style__Title>
      Uncle Vito's Pizza
    </style__Title>
    <style__RatingStarsWrapper>
      <style__StarIconWrapper
        key="0"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="1"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="2"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="3"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__StarIconWrapper
        key="4"
      >
        <StarIcon
          color="#E6E6E6"
        />
      </style__StarIconWrapper>
      <style__UserRatingsTotalText>
        (
        1427
        )
      </style__UserRatingsTotalText>
    </style__RatingStarsWrapper>
    <style__PriceLevelWrapper>
      <span>
        $
         
      </span>
      <span>
         
        •
         
      </span>
      <span>
         
        restaurant
      </span>
    </style__PriceLevelWrapper>
  </style__MetadataWrapper>
</style__HyperLinkWrapper>
`;
